pkg/boot/kexec/genpurg.go

```diff
@@ -0,0 +1,95 @@
+//go:build ignore
+
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func main() {
+	var fail bool
+	f, err := os.Create("asm.go")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			log.Fatal("Closing %v: %v", f.Name(), err)
+		}
+	}()
+
+	if _, err := fmt.Fprintf(f, "package kexec\nvar purgatories = map[string]*purgatory {\n"); err != nil {
+		log.Fatal(err)
+	}
+	d, err := ioutil.TempDir("", "kexecgen")
+	if err != nil {
+		log.Fatalf("%v", err)
+	}
+
+	for _, asm := range asms {
```

Comment:
I find it hard to read that this is defined in purgatories.go and we loop over it here. But that's personal preference.

Reply:
The assembly is going to be Go assembly at some point. I don't want to rely on the GNU toolchain in the long term.